src/core/observables/PidObservable.hpp

```diff
@@ -69,6 +69,30 @@ template <class T> struct shape_impl<std::vector<T>> {
     return ret;
   }
 };
+
+template <class OutputIterator> void flatten(double value, OutputIterator out) {
+  *out++ = value;
+}
+
+template <class OutputIterator>
+void flatten(Utils::Vector3d const &vec, OutputIterator out) {
+  for (auto const &elem : vec) {
+    flatten(elem, out);
+  }
+}
+
+inline std::vector<double> flatten(Utils::Vector3d const &vec) {
+  return vec.as_vector();
+}
+
+template <class T> std::vector<double> flatten(std::vector<T> const &vec) {
```

Comment:
This catches also a `std::vector<double>`, and then won't work. It solve this with SFINAE, a base case that just adds the value, and a specialization that loops and recurses. That would then also work for say `std::vector<Vector<Vector3>>`, e.g. a matrix.

Reply:
why wouldn't it work?

Reply:
Ah maybe it would